src/renderer/views/Watch/Watch.js

```diff
@@ -1266,6 +1264,23 @@ export default defineComponent({
       ]
     },
 
+    getAdaptiveFormatsInvidious: async function(existingInfoResult = null) {
+      let getInfoPromise = Promise.resolve(existingInfoResult)
+      if (existingInfoResult == null) {
+        getInfoPromise = invidiousGetVideoInformation(this.videoId)
+      }
+      const result = await getInfoPromise
```

Comment:
Is there a specific reason you decided to go with wrapping the result in a promise? If not, something like this seems less complex.

```suggestion
      let result
      if (existingInfoResult) {
        result = existingInfoResult
      } else {
        result = await invidiousGetVideoInformation(this.videoId)
      }
```

Reply:
This is me (still) not getting used to `await`/`async`